unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -2138,44 +2138,36 @@ handleUpdate input optionalPatch requestedNames = do
       fileNames :: Names
       fileNames = UF.typecheckedToNames uf
       -- todo: display some error if typeEdits or termEdits itself contains a loop
-      typeEdits :: Map Name (Reference, Reference)
-      typeEdits = Map.fromList $ map f (toList $ SC.types (updates sr))
-        where
-          f v = case ( toList (Names.typesNamed slurpCheckNames n),
-                       toList (Names.typesNamed fileNames n)
-                     ) of
-            ([old], [new]) -> (n, (old, new))
-            actual ->
-              error $
-                "Expected unique matches for var \""
-                  ++ Var.nameStr v
-                  ++ "\" but got: "
-                  ++ show actual
-            where
-              n = Name.unsafeFromVar v
+      typeEdits :: [(Name, Reference, Reference)]
+      typeEdits = do
+        v <- Set.toList (SC.types (updates sr))
+        let n = Name.unsafeFromVar v
+        let oldRefs0 = Names.typesNamed slurpCheckNames n
+        let newRefs = Names.typesNamed fileNames n
+        case (,) <$> NESet.nonEmptySet oldRefs0 <*> Set.asSingleton newRefs of
+          Nothing -> error (reportBug "E722145" ("bad (old,new) names: " ++ show (oldRefs0, newRefs)))
```

Comment:
Is the idea that we will see this error and then come back here to remind ourselves of what it means?

Reply:
Yeah, I guess so. (As opposed to what? :stuck_out_tongue:)